tests/integ_tests/polling_tests.rs

```diff
@@ -88,3 +98,125 @@ async fn out_of_order_completion_doesnt_hang() {
+    let server_config = TemporalDevServerConfigBuilder::default()
+        .exe(default_cached_download())
+        // We need to lower the poll timeout so the poll call rolls over
+        .extra_args(vec![
+            "--dynamic-config-value".to_string(),
+            "matching.longPollExpirationInterval=\"1s\"".to_string(),
+        ])
+        .build()
+        .unwrap();
+    let mut server1 = server_config.start_server().await.unwrap();
+    let mut server2 = server_config.start_server().await.unwrap();
+
+    // Connect clients to both servers
+    info!("Connecting clients");
+    let client1 = ClientOptionsBuilder::default()
+        .identity("integ_tester".to_owned())
+        .target_url(Url::parse(&format!("http://{}", server1.target)).unwrap())
+        .client_name("temporal-core".to_owned())
+        .client_version("0.1.0".to_owned())
+        .build()
+        .unwrap()
+        .connect("default", None)
+        .await
+        .unwrap();
+    let client2 = ClientOptionsBuilder::default()
+        .identity("integ_tester".to_owned())
+        .target_url(Url::parse(&format!("http://{}", server2.target)).unwrap())
+        .client_name("temporal-core".to_owned())
+        .client_version("0.1.0".to_owned())
+        .build()
+        .unwrap()
+        .connect("default", None)
+        .await
+        .unwrap();
+
+    // Start a workflow on both servers
+    info!("Starting workflows");
+    let wf1 = client1
+        .start_workflow(
+            vec![],
+            "my-task-queue".to_owned(),
+            "my-workflow-1".to_owned(),
+            "my-workflow-type".to_owned(),
+            None,
+            WorkflowOptions::default(),
+        )
+        .await
+        .unwrap();
+    let wf2 = client2
+        .start_workflow(
+            vec![],
+            "my-task-queue".to_owned(),
+            "my-workflow-2".to_owned(),
+            "my-workflow-type".to_owned(),
+            None,
+            WorkflowOptions::default(),
+        )
+        .await
+        .unwrap();
+
+    // Create a worker only on the first server
+    let worker = init_worker(
+        init_integ_telem(),
+        WorkerConfigBuilder::default()
+            .namespace("default")
+            .task_queue("my-task-queue")
+            .worker_build_id("test_build_id")
+            // We want a cache so we don't get extra remove-job activations
+            .max_cached_workflows(100_usize)
+            .build()
+            .unwrap(),
+        client1.clone(),
+    )
+    .unwrap();
+
+    // Poll for first task, confirm it's first wf, complete, and wait for complete
+    info!("Doing initial poll");
+    let act1 = worker.poll_workflow_activation().await.unwrap();
+    assert_eq!(wf1.run_id, act1.run_id);
+    worker
+        .complete_workflow_activation(WorkflowActivationCompletion::from_cmds(
+            act1.run_id,
+            vec![CompleteWorkflowExecution { result: None }.into()],
+        ))
+        .await
+        .unwrap();
```

Comment:
I have a shortcut for completing the execution you can use (`complete_execution`), or even just `from_cmd` rather than `from_cmds`

Reply:
Fixed